addons/migrations/README.md

```diff
@@ -1,9 +1,60 @@
+  task_definition_revision = 5 # The *new* revision to migrate *to*
+  min_capacity             = 0 # Scale down to this during migration
+  desired_count            = 2 # Scale back up to this after migration
+  subnets                  = ["subnet-xxxxxxxxxxxxxxxxx", "subnet-yyyyyyyyyyyyyyyyy"]
+  security_groups          = ["sg-xxxxxxxxxxxxxxxxx"]
+
+  # Optional: Specify if a separate vulnerability processing service needs coordination
+  # vuln_service             = "my-fleet-vuln-service"
+
+  # Optional: Provide an IAM Role ARN for the local-exec script to assume
+  # assume_role_arn          = "arn:aws:iam::123456789012:role/MyMigrationRole"
+  # assume_role_session_name = "TerraformFleetMigration"
+
+  # Ensure this module depends on the resource that creates/updates the task definition revision
+  # For example:
+  # depends_on = [aws_ecs_task_definition.fleet_app]
+}
+```
+
+## Workflow
+
+1.  When `var.task_definition_revision` changes, Terraform triggers the `null_resource`.
+2.  The `local-exec` provisioner executes the `migrate.sh` script located within the module's directory.
+3.  It passes essential AWS and ECS details (region, cluster, service, task definition, revision, network configuration, scaling parameters, optional role ARN) as command-line arguments or environment variables to the script.
+4.  The `migrate.sh` script (which you must provide and maintain) performs the migration steps against the Fleet database, using the provided parameters to interact with AWS ECS.
+
+## Prerequisites
+
+*   **`bash` shell:** Must be available in the environment where Terraform is executed.
+*   **AWS CLI:** Must be installed and configured with credentials in the environment where Terraform is executed. The credentials need permissions to perform ECS actions (DescribeServices, UpdateService, RunTask, DescribeTasks) and potentially STS AssumeRole if `assume_role_arn` is provided.
+*   **`migrate.sh` script:** A script named `migrate.sh` *must* exist within this module's directory (`path.module`). This script contains the actual logic for scaling services and running the migration task. **This module only triggers the script; it does not contain the migration logic itself.**
+*   **Existing Resources:** The specified ECS Cluster, Service, Task Definition (base ARN), Subnets, and Security Groups must exist.
+
+## Important Considerations
+
+*   **`local-exec`:** This provisioner runs commands on the machine executing Terraform. Ensure this machine has the necessary tools (bash, AWS CLI) and network access/credentials to interact with your AWS environment. This might require specific configuration in CI/CD pipelines.
+*   **IAM Permissions:** The credentials used by `local-exec` (either default AWS credentials or the assumed role specified by `assume_role_arn`) require sufficient IAM permissions to manage the specified ECS services and tasks.
```

Comment:
It would be great if we document some permission set to execute the migration script. Also which sources it should taget. Potentially this module could provide the role even? Given that the migrate.sh script can change and depends on certain resources.